spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -177,65 +181,67 @@ public boolean isSingleton() {
+			BoundElementHolder boundElementHolder = channelHolderEntry.getValue();
+			if (boundElementHolder.isBindable()) {
 				if (log.isDebugEnabled()) {
 					log.debug(String.format("Binding %s:%s:%s", this.channelNamespace, this.type, inputChannelName));
 				}
-				channelBindingService.bindConsumer(channelHolder.getMessageChannel(), inputChannelName);
+				bindingService.bindConsumer(boundElementHolder.getBoundElement(), inputChannelName);
 			}
 		}
 	}
 
 	@Override
-	public void bindOutputs(ChannelBindingService channelBindingService) {
+	public void bindOutputs(BindingService bindingService) {
 		if (log.isDebugEnabled()) {
 			log.debug(String.format("Binding outputs for %s:%s", this.channelNamespace, this.type));
 		}
-		for (Map.Entry<String, ChannelHolder> channelHolderEntry : this.outputHolders.entrySet()) {
-			ChannelHolder channelHolder = channelHolderEntry.getValue();
+		for (Map.Entry<String, BoundElementHolder> channelHolderEntry : this.outputHolders.entrySet()) {
+			BoundElementHolder boundElementHolder = channelHolderEntry.getValue();
 			String outputChannelName = channelHolderEntry.getKey();
 			if (channelHolderEntry.getValue().isBindable()) {
 				if (log.isDebugEnabled()) {
 					log.debug(String.format("Binding %s:%s:%s", this.channelNamespace, this.type, outputChannelName));
 				}
-				channelBindingService.bindProducer(channelHolder.getMessageChannel(), outputChannelName);
+				bindingService.bindProducer(boundElementHolder.getBoundElement(), outputChannelName);
 			}
 		}
 	}
 
 	@Override
-	public void unbindInputs(ChannelBindingService channelBindingService) {
+	public void unbindInputs(BindingService bindingService) {
 		if (log.isDebugEnabled()) {
 			log.debug(String.format("Unbinding inputs for %s:%s", this.channelNamespace, this.type));
 		}
-		for (Map.Entry<String, ChannelHolder> channelHolderEntry : this.inputHolders.entrySet()) {
+		for (Map.Entry<String, BoundElementHolder> channelHolderEntry : this.inputHolders.entrySet()) {
 			if (channelHolderEntry.getValue().isBindable()) {
 				if (log.isDebugEnabled()) {
-					log.debug(String.format("Unbinding %s:%s:%s", this.channelNamespace, this.type, channelHolderEntry.getKey()));
+					log.debug(String.format("Unbinding %s:%s:%s", this.channelNamespace, this.type,
+							channelHolderEntry.getKey()));
 				}
-				channelBindingService.unbindConsumers(channelHolderEntry.getKey());
+				bindingService.unbindConsumers(channelHolderEntry.getKey());
 			}
 		}
 	}
 
 	@Override
-	public void unbindOutputs(ChannelBindingService channelBindingService) {
+	public void unbindOutputs(BindingService bindingService) {
 		if (log.isDebugEnabled()) {
 			log.debug(String.format("Unbinding outputs for %s:%s", this.channelNamespace, this.type));
 		}
-		for (Map.Entry<String, ChannelHolder> channelHolderEntry : this.outputHolders.entrySet()) {
+		for (Map.Entry<String, BoundElementHolder> channelHolderEntry : this.outputHolders.entrySet()) {
```

Comment:
sure thing... I was just tracking them as I encountered, but definitely not necessary to address all of those changes within the scope of this PR if it's easier to submit a subsequent PR that addresses just those naming changes... and I can review it! ;)